packages/ui/src/components/vuestic-components/va-pagination/VaPagination.vue

```diff
@@ -1,215 +1,258 @@
     />
-    <va-button
-      :style="activeButtonStyle(n)"
-      outline
+    <slot v-if="!input">
+      <va-button
+        :style="activeButtonStyle(n)"
+        outline
+        v-for="(n, key) in paginationRange"
+        :key="key"
+        :color="color"
+        :size="size"
+        :disabled="disabled || n === '...'"
+        :class="{
+          'va-button--active': n === value,
+          'va-button--idle': n !== value,
+          'va-button--disabled': n === '...' || disabled,
+          'va-button--ellipsis': n === '...',
+        }"
+
+        @click="changePage(n)"
+        :flat="flat"
+      >
+        {{ n }}
+      </va-button>
+    </slot>
+    <va-input
+      v-else
       :color="color"
-      :size="size"
-      :disabled="disabled"
-      v-for="(n, key) in paginationRange"
-      :key="key"
-      :class="{ 'va-button--active': n === value }"
-      icon=""
-      icon-right=""
-      @click="changePage(n)"
-    >
-      {{ n }}
-    </va-button>
+      inside-pagination
+      ref="input"
+      class="va-button"
+      v-model="inputValue"
+      @keydown.enter="changeValue"
+      @focus="focusInput"
+      @blur="changeValue"
+      :placeholder="`${value}/${lastPage}`"
+    />
     <va-button
-      v-if="this.pages !== visiblePages && directionLinks"
-      :style="directionButtonStyle"
+      v-if="showDirectionLinks"
       outline
       :color="color"
       :size="size"
-      :disabled="disabled || value === this.pages"
+      :disabled="disabled || value === lastPage"
       :icon="iconRightClass.direction"
-      icon-right=""
       @click="changePage(value + 1)"
+      :flat="flat"
     />
     <va-button
-      v-if="this.pages !== visiblePages && boundaryLinks"
-      :style="directionButtonStyle"
+      v-if="showBoundaryLinks"
       outline
       :color="color"
       :size="size"
-      :disabled="disabled || value === this.pages"
+      :disabled="disabled || value === lastPage"
       :icon="iconRightClass.boundary"
-      icon-right=""
+      :flat="flat"
       @click="changePage(lastPage)"
     />
   </va-button-group>
 </template>
 
-<script>
-import VaButtonGroup from '../va-button-group/VaButtonGroup'
-import VaButton from '../va-button/VaButton'
+<script lang="ts">
+import VaButtonGroup from '../va-button-group/VaButtonGroup.vue'
+import VaButton from '../va-button/VaButton.vue'
+import VaInput from '../va-input/VaInput.vue'
+import { StatefulMixin } from '../../vuestic-mixins/StatefullMixin/StatefulMixin'
 import { setPaginationRange } from './setPaginationRange'
-import { ContextPluginMixin, getContextPropValue } from '../../context-test/context-provide/ContextPlugin'
+import { ContextPluginMixin, makeContextablePropsMixin } from '../../context-test/context-provide/ContextPlugin'
+import Component, { mixins } from 'vue-class-component'
+import { Watch } from 'vue-property-decorator'
+import { ColorThemeMixin } from '../../../services/ColorThemePlugin'
+
+interface IconSet {
```

Comment:
What's the point of this? Why not just use props with left and right icons?

Also IconSet is reserved.

Reply:
Make 4 props for icons